platform/features/table/src/TableConfiguration.js

```diff
@@ -144,16 +145,32 @@ define(
                 {}).columns || {};
         };
 
+        function equal(obj1, obj2) {
+            var obj1Keys = Object.keys(obj1),
+                obj2Keys = Object.keys(obj2);
+            return (obj1Keys.length === obj2Keys.length) &&
+                    obj1Keys.every(function(key){
+                        //To do a deep equals, could recurse here if typeof
+                        // obj1 === Object
+                       return obj1[key] === obj2[key];
+                    });
+        }
+
         /**
          * Set the established configuration on the domain object
          * @private
          */
         TableConfiguration.prototype.saveColumnConfiguration = function (columnConfig) {
```

Comment:
Interesting.  This is a private function but the only time it is called is by `TelemetryTableController.prototype.filterColumns`.

`TelemetryTableController.prototype.filterColumns` is never ever called with the columnConfig argument, but it conditionally checks for absence of the columnConfig argument (which, because it is never supplied) always evaluated true, and it runs a branch of code that fetches the configuration and then attempts to save it back.

The TelemetryTableController needs to be able to read the column configuration but there's no reason it needs to handle setting it.  This state could be internal to the TableConfiguration.